<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>868</width>
    <height>788</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hexagonal remainder range</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QDoubleSpinBox" name="doubleSpinBox">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>10</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1.047197000000000</double>
    </property>
    <property name="singleStep">
     <double>0.001000000000000</double>
    </property>
    <property name="value">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QLineEdit" name="x1">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>50</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="y1">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>50</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>130</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add points</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="y2">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>90</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="x2">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>90</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="ox">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>180</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="oy">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>180</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>220</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Change origin</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinBoxX">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>260</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
    <property name="value">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinBoxY">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>260</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
    <property name="value">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="pushBtnTrans">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>290</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Translation</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="hexTorusTab">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>741</width>
      <height>761</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Torus</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>330</y>
      <width>101</width>
      <height>321</height>
     </rect>
    </property>
    <property name="locale">
     <locale language="English" country="UnitedStates"/>
    </property>
    <property name="text">
     <string>Note that all the coordinates have to be given in the axial coordinate system. For instance, in the axial coordinate system the point (1,-1) corresponds to the point (sqrt(3)/2, 3/2) in the Cartesina coordinate system.</string>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
    <property name="margin">
     <number>-1</number>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>868</width>
     <height>31</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save torus</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changedAngle(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>655</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addPoints()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>654</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changedOrigin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>655</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveTorus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushBtnTrans</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changedTranslation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1185</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changedTranslation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1155</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changedTranslation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1215</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changedAngle(double)</slot>
  <slot>addPoints()</slot>
  <slot>changedOrigin()</slot>
  <slot>saveTorus()</slot>
  <slot>loadData()</slot>
  <slot>changedTranslation()</slot>
  <slot>changedAdj(int)</slot>
  <slot>saveGraph()</slot>
  <slot>exportTorusTiKZ()</slot>
  <slot>saveAnimation()</slot>
  <slot>saveAnimGraph()</slot>
  <slot>iterateOver()</slot>
 </slots>
</ui>
